src/Controllers/GagePageController.ts

```diff
@@ -190,6 +190,10 @@ module StreamStats.Controllers {
         public peakValuePlot: any;
         public annualFlowPlot: any;
         public peakValues: any;
+        public dischargeObj = undefined; // ethan
```

Comment:
Instead of using your name as comments maybe we can use the plot name, 'stage vs discharge plot'. This way we will know which variable is used by which plot.

Reply:
Not sure how to respond specifically to your second overall comment, so commenting here. The way I wrote USGSMeasured, there are color variables in there that appear every time whether there is data or not in it. So the showInLegend: this.measuredObj.length > 0 would not work since I was always getting data ("color"). So the way I wrote it: showInLegend: !this.measuredObj.every(item => isNaN(item.y)) . 

I went into the measuredObj array and said that if there is no y coordinate, then it won't show the legend. however I was seeing some random gauges that just had the X coord but not the Y. I didn't see any that just had a Y and not an X. Anyways, not sure if this is the right way to go about this.

Reply:
We can go with this for now and make changes later if we see that it's not working properly 👍

Reply:
And for future reference, I usually respond to overall comments by making a new comment at the very bottom of the Pull Request:

![image](https://user-images.githubusercontent.com/40237491/226746985-e503f702-3513-4b0b-9d72-d0943c78cfa4.png)

Reply:
I think my latest review will help make the USGSMeaured changes more robust, and you should be able to remove the code you added.